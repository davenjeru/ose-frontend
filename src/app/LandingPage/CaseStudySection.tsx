import type { FC, PropsWithChildren } from 'react';
import LandingPageIcon, {
  type LandingPageIconProps,
} from '@/lib/components/LandingPageIcon';
import Page from '@/lib/components/Page.tsx';
import financialDevastation from '@/lib/icons/financial-devastation.svg';
import moneyLoss from '@/lib/icons/money-loss.svg';
import operationalParalysis from '@/lib/icons/operational-paralysis.svg';

const CaseStudyStats: FC<
  PropsWithChildren<{ icon: LandingPageIconProps; title: string }>
> = ({ icon, title, children }) => (
  <div className="flex w-84 flex-col gap-4 font-sora">
    <LandingPageIcon {...icon} />
    <h1 className="text-left text-sora-heading-large text-secondary-foreground">
      {title}
    </h1>
    {children}
  </div>
);

const CaseStudySection = () => (
  <Page>
    <div className="-mt-45 flex flex-col items-center gap-16">
      <div className="flex flex-col items-center gap-4">
        <h1
          className={`
            text-center font-sora text-sora-heading-xxlarge font-light
          `}
        >
          Do You Remember{' '}
          <span className="text-secondary-foreground">Log4Shell?</span>
        </h1>
        <p
          className={`
            text-center font-roboto text-roboto-text-medium text-primary-grey
          `}
        >
          In December 2021, Log4Shell (CVE-2021-44228) exposed millions of
          Java-based systems <br /> worldwide. A single logging library
          triggered a global cybersecurity crisis with catastrophic <br />
          consequences.
        </p>
      </div>
      <div className="flex flex-row gap-8">
        <CaseStudyStats
          icon={{ src: moneyLoss, alt: 'Reputational Damage Icon' }}
          title="Reputational Damage"
        >
          <ul className="flex flex-col gap-4 font-roboto font-light">
            <li>Damaged brand reputation.</li>
            <li>Consumer trust erosion.</li>
            <li>Negative press and public scrutiny.</li>
          </ul>
        </CaseStudyStats>
        <CaseStudyStats
          icon={{
            src: financialDevastation,
            alt: 'Financial Devastation Icon',
          }}
          title="Financial Devastation"
        >
          <ul className="flex flex-col gap-4 font-roboto font-light">
            <li>
              <b>$700M+</b> legal exposure established.
            </li>
            <li>
              <b>$90,000+</b> average direct cost per breach
            </li>
            <li>Massive overtime for 24/7 “war rooms”.</li>
          </ul>
        </CaseStudyStats>
        <CaseStudyStats
          icon={{
            src: operationalParalysis,
            alt: 'Operational Paralysis Icon',
          }}
          title="Operational Paralysis"
        >
          <ul className="flex flex-col gap-4 font-roboto font-light">
            <li>
              <b>Up to 3,300 developer hours lost.</b>
            </li>
            <li>All innovation halted.</li>
            <li>29% recurrence rate forced costly rework cycles.</li>
          </ul>
        </CaseStudyStats>
      </div>
      <h3
        className={`
          font-sora text-sora-heading-small font-light text-primary-grey
        `}
      >
        Don’t let your organization be the next case study.
      </h3>
    </div>
  </Page>
);
export default CaseStudySection;
