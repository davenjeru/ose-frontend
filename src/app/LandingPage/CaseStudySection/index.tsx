import type { FC, PropsWithChildren } from 'react';
import LandingPageIcon, {
  type LandingPageIconProps,
} from '@/lib/components/LandingPageIcon';
import Page from '@/lib/components/Page.tsx';
import financialDevastation from '@/lib/icons/financial-devastation.svg';
import moneyLoss from '@/lib/icons/money-loss.svg';
import operationalParalysis from '@/lib/icons/operational-paralysis.svg';

const CaseStudyStats: FC<
  PropsWithChildren<{ icon: LandingPageIconProps; title: string }>
> = ({ icon, title, children }) => (
  <div
    className={`
      flex flex-col gap-4 text-center font-sora
      md:max-w-45 md:text-left
      xl:max-w-82
    `}
  >
    <LandingPageIcon
      {...icon}
      className={`
        self-center
        md:self-start
      `}
    />
    <h1
      className={`
        text-sora-heading-small text-secondary-foreground
        xl:text-sora-heading-large
      `}
    >
      {title}
    </h1>
    {children}
  </div>
);

const CaseStudySection = () => (
  <Page>
    <div className="flex flex-col items-center gap-16">
      <div className="flex flex-col items-center gap-4">
        <h1
          className={`
            text-center font-sora text-sora-heading-medium
            md:text-sora-heading-large md:font-light
            lg:text-sora-heading-xlarge
            xl:text-sora-heading-xxlarge
          `}
        >
          Do You Remember{' '}
          <span className="text-secondary-foreground">Log4Shell?</span>
        </h1>
        <p
          className={`
            text-center font-roboto text-roboto-text-small font-light
            text-primary-grey
            md:max-w-180 md:text-roboto-text-medium
          `}
        >
          In December 2021, Log4Shell (CVE-2021-44228) exposed millions of
          Java-based systems worldwide. A single logging library triggered a
          global cybersecurity crisis with catastrophic consequences.
        </p>
      </div>
      <div
        className={`
          flex w-full max-w-328 flex-col items-center gap-10 px-20
          md:flex-row md:justify-center md:gap-20
        `}
      >
        <CaseStudyStats
          icon={{ src: moneyLoss, alt: 'Reputational Damage Icon' }}
          title="Reputational Damage"
        >
          <ul
            className={`
              flex flex-col items-center gap-4 font-roboto font-light
              md:items-start
            `}
          >
            <li>Damaged brand reputation.</li>
            <li>Consumer trust erosion.</li>
            <li>Negative press and public scrutiny.</li>
          </ul>
        </CaseStudyStats>
        <CaseStudyStats
          icon={{
            src: financialDevastation,
            alt: 'Financial Devastation Icon',
          }}
          title="Financial Devastation"
        >
          <ul
            className={`
              flex flex-col items-center gap-4 font-roboto font-light
              md:items-start
            `}
          >
            <li>
              <b>$700M+</b> legal exposure established.
            </li>
            <li>
              <b>$90,000+</b> average direct cost per breach
            </li>
            <li>Massive overtime for 24/7 “war rooms”.</li>
          </ul>
        </CaseStudyStats>
        <CaseStudyStats
          icon={{
            src: operationalParalysis,
            alt: 'Operational Paralysis Icon',
          }}
          title="Operational Paralysis"
        >
          <ul
            className={`
              flex flex-col items-center gap-4 font-roboto font-light
              md:items-start
            `}
          >
            <li>
              <b>Up to 3,300 developer hours lost.</b>
            </li>
            <li>All innovation halted.</li>
            <li>29% recurrence rate forced costly rework cycles.</li>
          </ul>
        </CaseStudyStats>
      </div>
      <h3
        className={`
          text-center font-sora text-sora-heading-xsmall font-light
          text-primary-grey
          md:text-sora-heading-small
        `}
      >
        Don’t let your organization be the next case study.
      </h3>
    </div>
  </Page>
);
export default CaseStudySection;
