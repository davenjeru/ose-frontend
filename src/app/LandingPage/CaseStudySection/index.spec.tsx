import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import CaseStudySection from '@/app/LandingPage/CaseStudySection/index.tsx';

describe('CaseStudySection', () => {
  it('renders without crashing', () => {
    const { container } = render(<CaseStudySection />);
    expect(container).toMatchSnapshot();
  });
});
