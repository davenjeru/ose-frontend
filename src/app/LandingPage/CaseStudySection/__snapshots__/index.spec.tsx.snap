// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CaseStudySection > renders without crashing 1`] = `
<div>
  <div
    class="mb-40 bg-background px-4 text-primary-foreground"
  >
    <div
      class="flex flex-col items-center gap-16"
    >
      <div
        class="flex flex-col items-center gap-4"
      >
        <h1
          class="
            text-center font-sora text-sora-heading-medium
            md:text-sora-heading-large md:font-light
            lg:text-sora-heading-xlarge
            xl:text-sora-heading-xxlarge
          "
        >
          Do You Remember
           
          <span
            class="text-secondary-foreground"
          >
            Log4Shell?
          </span>
        </h1>
        <p
          class="
            text-center font-roboto text-roboto-text-small font-light
            text-primary-grey
            md:max-w-180 md:text-roboto-text-medium
          "
        >
          In December 2021, Log4Shell (CVE-2021-44228) exposed millions of Java-based systems worldwide. A single logging library triggered a global cybersecurity crisis with catastrophic consequences.
        </p>
      </div>
      <div
        class="
          flex w-full max-w-328 flex-col items-center gap-10 px-20
          md:flex-row md:justify-center md:gap-20
        "
      >
        <div
          class="
      flex flex-col gap-4 text-center font-sora
      md:max-w-45 md:text-left
      xl:max-w-82
    "
        >
          <img
            alt="Reputational Damage Icon"
            class="self-center md:self-start"
            height="40"
            src="data:image/svg+xml,%3csvg%20width='40'%20height='40'%20viewBox='0%200%2040%2040'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M34.9999%2018.3333C34.5579%2018.3333%2034.134%2018.5089%2033.8214%2018.8215C33.5088%2019.134%2033.3332%2019.558%2033.3332%2020V24.3166L22.8499%2013.8167C22.695%2013.6604%2022.5106%2013.5364%2022.3075%2013.4518C22.1044%2013.3672%2021.8866%2013.3237%2021.6666%2013.3237C21.4465%2013.3237%2021.2287%2013.3672%2021.0256%2013.4518C20.8225%2013.5364%2020.6382%2013.6604%2020.4832%2013.8167L14.9999%2019.3167L6.18324%2010.4833C5.8694%2010.1695%205.44374%209.99316%204.9999%209.99316C4.55607%209.99316%204.13041%2010.1695%203.81657%2010.4833C3.50273%2010.7972%203.32642%2011.2228%203.32642%2011.6667C3.32642%2012.1105%203.50273%2012.5361%203.81657%2012.85L13.8166%2022.85C13.9715%2023.0062%2014.1558%2023.1302%2014.3589%2023.2148C14.562%2023.2994%2014.7799%2023.343%2014.9999%2023.343C15.2199%2023.343%2015.4378%2023.2994%2015.6409%2023.2148C15.844%2023.1302%2016.0283%2023.0062%2016.1832%2022.85L21.6666%2017.35L30.9832%2026.6666H26.6666C26.2245%2026.6666%2025.8006%2026.8422%2025.4881%2027.1548C25.1755%2027.4674%2024.9999%2027.8913%2024.9999%2028.3333C24.9999%2028.7753%2025.1755%2029.1993%2025.4881%2029.5118C25.8006%2029.8244%2026.2245%2030%2026.6666%2030H34.9999C35.2177%2029.9973%2035.4329%2029.9521%2035.6332%2029.8666C36.0405%2029.6975%2036.3641%2029.3739%2036.5332%2028.9666C36.6186%2028.7663%2036.6639%2028.5511%2036.6666%2028.3333V20C36.6666%2019.558%2036.491%2019.134%2036.1784%2018.8215C35.8659%2018.5089%2035.4419%2018.3333%2034.9999%2018.3333Z'%20fill='%23FF8AB2'/%3e%3c/svg%3e"
            width="40"
          />
          <h1
            class="
        text-sora-heading-small text-secondary-foreground
        xl:text-sora-heading-large
      "
          >
            Reputational Damage
          </h1>
          <ul
            class="
              flex flex-col items-center gap-4 font-roboto font-light
              md:items-start
            "
          >
            <li>
              Damaged brand reputation.
            </li>
            <li>
              Consumer trust erosion.
            </li>
            <li>
              Negative press and public scrutiny.
            </li>
          </ul>
        </div>
        <div
          class="
      flex flex-col gap-4 text-center font-sora
      md:max-w-45 md:text-left
      xl:max-w-82
    "
        >
          <img
            alt="Financial Devastation Icon"
            class="self-center md:self-start"
            height="40"
            src="data:image/svg+xml,%3csvg%20width='40'%20height='40'%20viewBox='0%200%2040%2040'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M5%2020C5%2021.9698%205.38799%2023.9204%206.14181%2025.7403C6.89563%2027.5601%208.00052%2029.2137%209.3934%2030.6066C10.7863%2031.9995%2012.4399%2033.1044%2014.2597%2033.8582C16.0796%2034.612%2018.0302%2035%2020%2035C21.9698%2035%2023.9204%2034.612%2025.7403%2033.8582C27.5601%2033.1044%2029.2137%2031.9995%2030.6066%2030.6066C31.9995%2029.2137%2033.1044%2027.5601%2033.8582%2025.7403C34.612%2023.9204%2035%2021.9698%2035%2020C35%2018.0302%2034.612%2016.0796%2033.8582%2014.2597C33.1044%2012.4399%2031.9995%2010.7863%2030.6066%209.3934C29.2137%208.00052%2027.5601%206.89563%2025.7403%206.14181C23.9204%205.38799%2021.9698%205%2020%205C18.0302%205%2016.0796%205.38799%2014.2597%206.14181C12.4399%206.89563%2010.7863%208.00052%209.3934%209.3934C8.00052%2010.7863%206.89563%2012.4399%206.14181%2014.2597C5.38799%2016.0796%205%2018.0302%205%2020Z'%20stroke='%23FF8AB2'%20stroke-width='3'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3cpath%20d='M24.6667%2015C24.3647%2014.4762%2023.9261%2014.0445%2023.3976%2013.7509C22.8692%2013.4573%2022.2708%2013.313%2021.6667%2013.3333H18.3333C17.4493%2013.3333%2016.6014%2013.6845%2015.9763%2014.3096C15.3512%2014.9347%2015%2015.7826%2015%2016.6666C15%2017.5507%2015.3512%2018.3985%2015.9763%2019.0236C16.6014%2019.6488%2017.4493%2020%2018.3333%2020H21.6667C22.5507%2020%2023.3986%2020.3511%2024.0237%2020.9763C24.6488%2021.6014%2025%2022.4492%2025%2023.3333C25%2024.2173%2024.6488%2025.0652%2024.0237%2025.6903C23.3986%2026.3154%2022.5507%2026.6666%2021.6667%2026.6666H18.3333C17.7292%2026.687%2017.1308%2026.5426%2016.6024%2026.249C16.0739%2025.9555%2015.6353%2025.5237%2015.3333%2025M20%2011.6666V28.3333'%20stroke='%23FF8AB2'%20stroke-width='3'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3c/svg%3e"
            width="40"
          />
          <h1
            class="
        text-sora-heading-small text-secondary-foreground
        xl:text-sora-heading-large
      "
          >
            Financial Devastation
          </h1>
          <ul
            class="
              flex flex-col items-center gap-4 font-roboto font-light
              md:items-start
            "
          >
            <li>
              <b>
                $700M+
              </b>
               legal exposure established.
            </li>
            <li>
              <b>
                $90,000+
              </b>
               average direct cost per breach
            </li>
            <li>
              Massive overtime for 24/7 “war rooms”.
            </li>
          </ul>
        </div>
        <div
          class="
      flex flex-col gap-4 text-center font-sora
      md:max-w-45 md:text-left
      xl:max-w-82
    "
        >
          <img
            alt="Operational Paralysis Icon"
            class="self-center md:self-start"
            height="40"
            src="data:image/svg+xml,%3csvg%20width='40'%20height='40'%20viewBox='0%200%2040%2040'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M16.6668%2011.6667C16.2248%2011.6667%2015.8009%2011.8423%2015.4883%2012.1549C15.1758%2012.4674%2015.0002%2012.8913%2015.0002%2013.3334V26.6667C15.0002%2027.1087%2015.1758%2027.5327%2015.4883%2027.8452C15.8009%2028.1578%2016.2248%2028.3334%2016.6668%2028.3334C17.1089%2028.3334%2017.5328%2028.1578%2017.8453%2027.8452C18.1579%2027.5327%2018.3335%2027.1087%2018.3335%2026.6667V13.3334C18.3335%2012.8913%2018.1579%2012.4674%2017.8453%2012.1549C17.5328%2011.8423%2017.1089%2011.6667%2016.6668%2011.6667ZM20.0002%203.33337C16.7038%203.33337%2013.4815%204.31086%2010.7407%206.14221C7.99985%207.97357%205.86364%2010.5765%204.60218%2013.622C3.34072%2016.6674%203.01066%2020.0185%203.65375%2023.2515C4.29684%2026.4846%205.88418%2029.4543%208.21506%2031.7852C10.5459%2034.116%2013.5156%2035.7034%2016.7487%2036.3465C19.9817%2036.9896%2023.3328%2036.6595%2026.3782%2035.398C29.4237%2034.1366%2032.0266%2032.0004%2033.858%2029.2595C35.6894%2026.5187%2036.6668%2023.2964%2036.6668%2020C36.6668%2017.8113%2036.2357%2015.6441%2035.3982%2013.622C34.5606%2011.5999%2033.3329%209.76257%2031.7853%208.21493C30.2376%206.66728%2028.4003%205.43963%2026.3782%204.60205C24.3561%203.76447%2022.1889%203.33337%2020.0002%203.33337ZM20.0002%2033.3334C17.3631%2033.3334%2014.7852%2032.5514%2012.5926%2031.0863C10.3999%2029.6212%208.69095%2027.5388%207.68178%2025.1025C6.67261%2022.6661%206.40856%2019.9853%206.92303%2017.3988C7.4375%2014.8124%208.70738%2012.4367%2010.5721%2010.572C12.4368%208.70725%2014.8126%207.43737%2017.399%206.9229C19.9854%206.40843%2022.6663%206.67248%2025.1026%207.68165C27.539%208.69082%2029.6213%2010.3998%2031.0864%2012.5924C32.5515%2014.7851%2033.3335%2017.363%2033.3335%2020C33.3335%2023.5363%2031.9287%2026.9276%2029.4283%2029.4281C26.9278%2031.9286%2023.5364%2033.3334%2020.0002%2033.3334ZM23.3335%2011.6667C22.8915%2011.6667%2022.4676%2011.8423%2022.155%2012.1549C21.8424%2012.4674%2021.6668%2012.8913%2021.6668%2013.3334V26.6667C21.6668%2027.1087%2021.8424%2027.5327%2022.155%2027.8452C22.4676%2028.1578%2022.8915%2028.3334%2023.3335%2028.3334C23.7755%2028.3334%2024.1995%2028.1578%2024.512%2027.8452C24.8246%2027.5327%2025.0002%2027.1087%2025.0002%2026.6667V13.3334C25.0002%2012.8913%2024.8246%2012.4674%2024.512%2012.1549C24.1995%2011.8423%2023.7755%2011.6667%2023.3335%2011.6667Z'%20fill='%23FF8AB2'/%3e%3c/svg%3e"
            width="40"
          />
          <h1
            class="
        text-sora-heading-small text-secondary-foreground
        xl:text-sora-heading-large
      "
          >
            Operational Paralysis
          </h1>
          <ul
            class="
              flex flex-col items-center gap-4 font-roboto font-light
              md:items-start
            "
          >
            <li>
              <b>
                Up to 3,300 developer hours lost.
              </b>
            </li>
            <li>
              All innovation halted.
            </li>
            <li>
              29% recurrence rate forced costly rework cycles.
            </li>
          </ul>
        </div>
      </div>
      <h3
        class="
          text-center font-sora text-sora-heading-xsmall font-light
          text-primary-grey
          md:text-sora-heading-small
        "
      >
        Don’t let your organization be the next case study.
      </h3>
    </div>
  </div>
</div>
`;
